.buildkite/filebeat/scripts/package.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+source .buildkite/filebeat/scripts/common.sh
+
+echo ":: Evaluate Filebeat Changes ::"
+
+changeset="^filebeat/
+    ^go.mod
+    ^pytest.ini
+    ^dev-tools/
+    ^libbeat/
+    ^testing/
+    ^\.buildkite/filebeat/"
+
+if ! are_files_changed "$changeset" ; then
```

Comment:
IMO, the change-set validation should not run in the `package.sh` but earlier. Otherwise, the package steps will always run but do nothing if no change-set changes. That will cause some downsides:
- Request a VM for installing specific dependencies, such as python3 and so on - so chances that the package manager can fail.
- Add a Buildkite warning if there are no changes - add warnings in the UI to say it didn't run. It's a bit of overhead.
- Cost in the Cloud provider and Buildkite.

Reply:
There is no "earlier" stage where we can validate the change-set, because the "if" statement is being parsed/analyzed on the pipeline upload stage, which can't modified. All other hooks & scripts are executed afterwards - so there is no possible option to validate the the change-set before the packaging script execution (e.g. export the result to env var and the use it as step execution condition).
As mentioned before, we will create beats-specific images and dependency install won't be required and pipelines will be updated accordingly.

Reply:
What about using [dynamic pipelines](https://buildkite.com/docs/pipelines/defining-steps#dynamic-pipelines)?

Something like:

```bash
# .buildkite/package.sh
set -eu

if are_files_changed "$changeset" ; then
    cat <<-EOF
  - group: "Packaging"
    key: "packaging"
    if: build.env("BUILDKITE_PULL_REQUEST") != "false"
    depends_on:
      - "mandatory-tests"
      - "extended-tests"
      - "extended-tests-win"

    steps:
      - label: ":ubuntu: Packaging Linux X86"
        key: "package-linux-x86"
        command:
          - "sudo apt-get update"
          - "sudo apt-get install -y python3-venv"
          - ".buildkite/filebeat/scripts/package.sh"
        notify:
          - github_commit_status:
              context: "Filebeat/Packaging: Linux X86"
        agents:
          provider: "gcp"
          image: "${IMAGE_UBUNTU_X86_64}"

      - label: ":linux: Packaging Linux ARM"
        key: "package-linux-arm"
        env:
          PLATFORMS: "linux/arm64"
          PACKAGES: "docker"
        command:
          - "sudo apt-get update"
          - "sudo apt-get install -y python3-venv"
          - ".buildkite/filebeat/scripts/package.sh"
        notify:
          - github_commit_status:
              context: "Filebeat/Packaging: ARM"
        agents:
          provider: "aws"
          imagePrefix: "${IMAGE_UBUNTU_ARM_64}"
          instanceType: "t4g.large"
EOF
fi
```

```yaml
steps:
  ...
  - label: Package pipeline
    commands:
    - .buildkite/package.sh | buildkite-agent pipeline upload
```